### PR TITLE
Add multiple files per snippet

### PR DESCRIPTION
**Pull request by GWillmann**
What an awesome project Snibox is! I hope my contribution make it even better.

This PR adds support of **multiple files per snippet**. 
We needed this kind of feature at Kinoba and I saw that [several people wanted it also](https://github.com/snibox/snibox/issues/6#issue-338351375)

I added the following model: `SnippetFile` which belong_to a `Snippet`.
Also I was inspired by GitHub's snippet system and I replace the Snippet title by a description.

I also changed the tests according to the new structure and added new ones:

- Add a snippet file
- Remove a snippet file

I am not extremely happy with the design I came up when one edits a `Snippet`, if you guys have some suggestions I'd be happy to change it.

### CODE REVIEW

**Comment by GWillmann** [comment]
Hello @vavgustov,

Have you had the time to review my PR? 

Let me know if I can improve something :)

**Comment by vavgustov** [comment]
hi @GWillmann ,

I apologize for the delay, I have very limited time now. I'll check your PR at 04/21.

**Comment by vavgustov** [comment]
hi,

just did a quick look at your PR. **Great job!** :godmode: 

Some quick recommendations about UI and other things that I noticed:

1. Current second column from existing UI not optimized for long descriptions. 
I think we need both title + description for snippet.
That means 20190401062316_convert_snippets_to_snippet_files.rb should't remove title field from snippets table. 
Title should be required and description should be optional.

2. 'Edit mode': 'Add snippet file' => should be just one button (at this moment this button under each file). Name  should be 'Add file'.

3. Files probably should be collapsible?

4. 'Show mode': files should't have 'delete' button.

5. 'Edit mode': labels should be moved above files.

6. 'Edit mode': file textfield and other UI elements should be moved to header? Delete button should stick to textfield as icon?

Probably a lot of other small UI improvements :)

I didn't had enough time so I just did a quick draft for some changes for 'show' mode:
show_mode

**PS** this is not final recommendations. I need more time to check this big PR in more detail and then
 provide detailed suggestions/screenshots at 04/28 or 05/05. Sorry for the delays and thanks for your patience!

**Comment by GWillmann** [comment]
Hi @vavgustov,

thanks for your feedback!

I have taken them into account in my PR. Quick info:

## Commit 98779e6

Removes `chromedriver-helper` gem because it is now deprecated in favor of the `webdrivers` gem

## Commit bf4d058

1) I readded the title attribute to the Snippet model and made the description attribute optional
2) `Add snippet file` button was renamed to `Add file` and only one occurence of the button was kept
3) Snippet files are now callapsible in show and edit mode
4) I removed the `delete` button in show mode for snippet files
5) I moved labels above the snippet files in create/edit mode

I also added tests for snippet files callapsible capability

Finally I am not sure I understood correctly your comment: 

> 6 - 'Edit mode': file textfield and other UI elements should be moved to header? Delete button should stick to textfield as icon?

Could you rephrase it please?

**Comment by vavgustov** [comment]
awesome, thanks!

don't worry about 6, I'll take care of it.

also I'll make some experiments with existing UI soon and commit them directly to your PR (few small changes already added).

**Comment by vavgustov** [comment]
@GWillmann when you have time can you replace `underscore_case` to `camelCase` for js code?

e.g. `snippet_files` should be `snippetFiles`
or `destroy_snippet_file` should be `destroySnippetFile`

Exception: requests to backend, e.g. it's ok to have:
`snippet_files_attributes: snippetFilesAttributes`,
because we use `underscore_case` for model properties (rails).

for `ActiveModel::Serializer` response you can use something like this: https://stackoverflow.com/a/40618505 so the server can respond directly with `snippetFiles` instead of `snippet_files`.

PS don't forget to pull the changes for this PR :)

**Comment by GWillmann** [comment]
I have changed made the changes you requested @vavgustov.

Do you think you will be able to merge this PR soon? :)

**Comment by vavgustov** [comment]
thanks!
It will be merged during May.

**Comment by vavgustov** [comment]
ok I added some changes and merged to master. I would like to make more but don't have a time as usual :smiley:. Thanks for your efforts!
